File: 136. Single Number/136-single-number.java
// 136. Single Number
// https://leetcode.com/problems/single-number/description/
class Solution {
    public int singleNumber(int[] nums) {
        int result = 0; // Initialize result to 0
        for (int num : nums){ // Loop through each number in the array
            result ^= num; // Use XOR operation to find the single number
        }
        return result; // Return the result
    }
}

File: NeetCode 150/136-Single Number.py
# Leetcode Problem 136: Single Number
# https://leetcode.com/problems/single-number/description/
from typing import List
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # if the list is empty, we return 0
        if not nums:
            return 0
        # if the length of the list is 1, we return that number
        if len(nums) == 1:
            return nums[0]
        # Initialize an empty dictionary to find the frequency of each number
        hash_map = {}
        # find the frequency of each number from the list
        for num in nums:
            hash_map[num] = hash_map.get(num, 0) + 1
        # loop through the dictionary and find whose count is less than 2 and return that number
        for num, count in hash_map.items():
            if count < 2:
                return num

print("Time Complexity: O(N)")
print("Space Complexity: O(N)")